lib/sidekiq/grouping/redis.rb

```diff
@@ -91,6 +122,12 @@ def push_msg(name, msg, remember_unique = false)
         end
       end
 
+      def push_messages(name, messages, remember_unique = false)
+        keys = [ns('batches'), name, ns(name), unique_messages_key(name), remember_unique]
+        args = [messages]
+        redis { |conn| conn.evalsha @script_hashes[:merge_array], keys, args }
```

Comment:
Running a script is an exclusive operation, and messages is unbounded here. I suggest we set a hard limit on the size of messages and batch them accordingly. Sidekiq does a similar thing with `push_bulk` (although it doesn't use lua scripting for pushes). If we don't batch, an unbounded push will block all redis requests for an indeterminate amount of time.

https://github.com/mperham/sidekiq/blob/main/lib/sidekiq/job.rb#L240-L247

Reply:
Great point. Thanks for pointing this out. I put the batching of these inserts up a level at the batch itself but that wouldn't stop someone from misusing it here. Going to push another commit to move that here instead soon.